azurerm/resource_arm_eventgrid_topic.go

```diff
@@ -0,0 +1,154 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/arm/eventgrid"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmEventGridTopic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmEventGridTopicCreateUpdate,
+		Read:   resourceArmEventGridTopicRead,
+		Update: resourceArmEventGridTopicCreateUpdate,
+		Delete: resourceArmEventGridTopicDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"tags": tagsSchema(),
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_access_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"secondary_access_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmEventGridTopicCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).eventGridTopicsClient
+
+	log.Printf("[INFO] preparing arguments for AzureRM EventGrid Topic creation.")
```

Comment:
I think this log message could be a lot more useful if it was moved down to line 71 and also displayed contents of `properties` before passing it to the SDK.